tests/cupy_tests/linalg_tests/test_norms.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+
+import numpy
+
+from cupy import testing
+
+
+@testing.gpu
+class TestTrace(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_trace(self, xp, dtype):
+        a = testing.shaped_arange((2, 3, 4, 5), xp, dtype)
+        return a.trace(1, 3, 2)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_external_trace(self, xp, dtype):
+        a = testing.shaped_arange((2, 3, 4, 5), xp, dtype)
+        return xp.trace(a, 1, 3, 2)
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(1,), (2,)],
+    'ord': [-numpy.Inf, -2, -1, 0, 1, 2, 3, numpy.Inf],
+    'axis': [0, None],
+    'keepdims': [True, False],
+}) + testing.product({
+    'shape': [(1, 2), (2, 2)],
+    'ord': [-numpy.Inf, -1, 1, numpy.Inf, 'fro'],
+    'axis': [(0, 1), None],
+    'keepdims': [True, False],
+}) + testing.product({
+    'shape': [(2, 2, 2)],
+    'ord': [-numpy.Inf, -2, -1, 0, 1, 2, 3, numpy.Inf],
+    'axis': [0, 1, 2],
+    'keepdims': [True, False],
+}) + testing.product({
+    'shape': [(2, 2, 2)],
+    'ord': [-numpy.Inf, -1, 1, numpy.Inf, 'fro'],
+    'axis': [(0, 1), (0, 2), (1, 2)],
+    'keepdims': [True, False],
+})
+)
+@testing.gpu
+class TestNorm(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-4, type_check=False)
```

Comment:
Given `float32` array as an in input, numpy norm function returns sometimes `flaot32` and sometimes `float64`. For example,

```python
>>> import numpy
>>> x = numpy.array([1.1, 1.2], dtype=numpy.float32)
>>> numpy.linalg.norm(x, ord=2).dtype
dtype('float32')
>>> numpy.linalg.norm(x, ord=3).dtype
```